src/UhdmAst.cpp

```diff
@@ -2538,215 +2553,231 @@ AstNode* visit_object(vpiHandle obj_h, UhdmShared& shared) {
+            VL_SIMPLE_SYSFUNC_CALL($fwriteo, Display, AstDisplayType::DT_WRITE, args[0], vectorToAstList(args.begin() + 1, args.end()), 'o'),
+            VL_SIMPLE_SYSFUNC_CALL($info, Display, AstDisplayType::DT_INFO, nullptr, args[0]),
+            VL_SIMPLE_SYSFUNC_CALL($warning, Display, AstDisplayType::DT_WARNING, nullptr, args[0]),
+            VL_SYSFUNC_CALL($error, {
+                    auto* displayp = new AstDisplay(fl, AstDisplayType::DT_ERROR, nullptr, vectorToAstList(args));
+                    displayp->addNext(new AstStop(fl, true));
+                    return displayp;
+                }),
+            VL_SYSFUNC_CALL($fatal, {
+                    auto* displayp = new AstDisplay(fl, AstDisplayType::DT_FATAL, nullptr, vectorToAstList(args));
+                    displayp->addNext(new AstStop(fl, false));
+                    return displayp;
+                }),
+            VL_SIMPLE_SYSFUNC_CALL($monitoroff, MonitorOff, true),
+            VL_SIMPLE_SYSFUNC_CALL($monitoron, MonitorOff, false),
+            VL_SYSFUNC_CALL($printtimescale, { return new AstPrintTimeScale(fl); }),
+            VL_SIMPLE_SYSFUNC_CALL($timeformat, TimeFormat, args[0], args[1], args[2], args[3]),
+            VL_SIMPLE_SYSFUNC_CALL($readmemb, ReadMem, false, args[0], args[1], args.size() > 2 ? args[2] : nullptr, args.size() > 3 ? args[3] : nullptr),
+            VL_SIMPLE_SYSFUNC_CALL($readmemh, ReadMem, true, args[0], args[1], args.size() > 2 ? args[2] : nullptr, args.size() > 3 ? args[3] : nullptr),
+            VL_SIMPLE_SYSFUNC_CALL($writememb, WriteMem, false, args[0], args[1], args.size() > 2 ? args[2] : nullptr, args.size() > 3 ? args[3] : nullptr),
+            VL_SIMPLE_SYSFUNC_CALL($writememh, WriteMem, true, args[0], args[1], args.size() > 2 ? args[2] : nullptr, args.size() > 3 ? args[3] : nullptr),
+            VL_SIMPLE_SYSFUNC_CALL($cast, CastParse, args[0], args[1]),
+
+            VL_SYSFUNC_CALL($fopen, {
+                    auto parent_h = vpi_handle({vpiParent}, obj_h);
+                    auto lhs_h = vpi_handle({vpiLhs}, parent_h);
+                    AstNode* fd = nullptr;
+                    if (lhs_h) { fd = visit_object(lhs_h, shared); }
+                    return new AstFOpen(make_fileline(obj_h), fd, args[0], args[1]);
+                }),
+            VL_SYSFUNC_CALL($__BAD_SYMBOL__, {
+                    fl->v3info("Bad symbol encountered!");
+                    // Dummy statement to keep parsing
+                    return new AstTime(make_fileline(obj_h), VTimescale::NONE);
+                })
+        };
+        auto it = sysFuncCallMakers.find(objectName);
+        if (it != sysFuncCallMakers.end())
+            node = it->second(obj_h, shared, arguments);
+        else
             v3error("\t! Encountered unhandled SysFuncCall: " << objectName);
+        if (VN_IS(node, NodeMath) && !VN_IS(node, Signed) && !VN_IS(node, Unsigned)) {
+            auto parent_h = vpi_handle(vpiParent, obj_h);
```

Comment:
This handle is never released.

Reply:
If `parent_type == vpiBegin` the handle still won't be released. Please move `vpi_release_handle(parent_h);` before that `if`.

Reply:
Oops. Fixed.